### DISCUSSION TITLE
Unbreakable anvils

### DISCUSSION THREAD

**Discussion by SinsOfSolitude** [Zrips/CMI#3753] [Feature Requests]
Feature to make anvils unbreakable when used

---

Theres should be a command to make anvils not break or get automatically replaced by a new anvil if broken.
Command example "/cmi disableanvilbreak [all/playername] [true/false] [world] [x;y;z]
If set to true, anvils do not break/get automatically replaced when they reach zero durability. If you select world, anvils wont break or get replaced on break in selected world. If you select coordinates, the anvil in x:y:z coordinate will be unbreakable or gets replaced on break.

---

The reason this feature is necessary is because the only way to currently do this is you set an auto-execute command when pressing on an anvil, to use the /anvil gui instead, or disable the anvil break event thats built into minecraft, by yourself, and most players dont even know such a thing is possible. I've read a lot about people having this issue and not having any plugins to fix this issue, and if there are plugins, they are always out-dated. A lot of servers create a custom plugin for themself which does this, as most server styles require anvils to never break from using them. Thank you <3

**Comment by impmallet** [comment]
This would be a very neat feature!

In the mean time, if you want to imitate this functionality, you can attach an interactive command to the anvil that you want to be permanent, and then have it issue the /cmi anvil command to the player who right clicks it. Then just make sure your anvil is protected with worldguard or what not and you're good to go.

**Comment by SinsOfSolitude** [comment]
Yeah, I've already done that as an alternative, doesn't feel super reliable but yeah, it is what it is. Hopefully this feature will atleast be taken into consideration..

**Comment by CloudeLecaw** [comment]
What happens if you give an item the unbreakeable option from cmi and place it?

Hold anvil in hand and use
/Cmi unbreakable (playerName) true
Makes item unbreakable.. 

Maybe it does not work for anvils? If not this would be a nice add to this feature. 

Have you tested this allready?

**Comment by SinsOfSolitude** [comment]
I have indeed tried that and unfortunately that gives the unbreakable feature only to usable tools that have durability. Did not work on an anvil after placing it :l

**Comment by CloudeLecaw** [comment]
May Zrips could add this as a feature for Anvil :) would make sense since the item is allready labeld as "unbreaklable" when holding it

**Comment by SinsOfSolitude** [comment]
I'm fairly certain this feature would be super easy to add as well, just have to find the listener for anvil durability or breaking, and then via command disable said listener. Hopefully it'll be added. A LOT of servers need unbreakable anvils at spawns and so on, so most of them have to code a plugin for this themselves or hire a plugineer to make it for them. I bet this feature alone would bring quite a few new people into using CMI.

**Comment by 456dev** [comment]
ok, ive looked into it quickly, and i cant see an native bukkit/spigot event for anvil damage, so the only way i say was packets (intercepting the anvilbreak sound packet?)

**Comment by SinsOfSolitude** [comment]
I've read about it before posting this, and yes, there aint an exact event for this on bukkit or spigot, though it should be possible to use the block break event targeted at anvils exclusively. Another alternative would be something to check anvil states, so that anvils should always be full durability, as soon as an anvil thats selected with CMI, gets ''slightly damaged'', its automatically replaced with a completely new anvil. The best option I could think of, would be to fully disable anvil GUI and replace it with cmi's anvil GUI, and have a config option for doing this globally or with select blocks. Yes you can already do this with interactiveCommands, but thats called a workaround, not a feature. There are a lot of options other than just disabling an event.
